### PR TITLE
W1 - D4 - Event - Listener

### PR DESCRIPTION
**Pull request by at-baole**

### CODE REVIEW

**Comment by at-baole** [comment]
@at-vinhhuynh please review it, thanks.

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import java.util.regex.Pattern;
+
+public class SignUpActivity extends AppCompatActivity {
+
+    EditText mUsername;
+    EditText mPassword;
+    EditText mEmail;
+    RadioButton mRadioButton1;
+    RadioButton mRadioButton2;
+    ImageButton mApply;
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: change variable name flow convention

Comment (at-vinhhuynh) [reply app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: add `private`

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    private boolean mRadioChecked;
+
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: remove this line

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_sign_up);
+        mUsername = findViewById(R.id.edtUsername);
+        mPassword = findViewById(R.id.edtPassword);
+        mEmail = findViewById(R.id.edtEmail);
+        mRadioButton1 = findViewById(R.id.rbMale);
+        mRadioButton2 = findViewById(R.id.rbFemale);
+        mApply = findViewById(R.id.btnApply);
+        mApply.setEnabled(false);
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: remove this line and set on xml file

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                //Check the length of Username, it must contain at least 6 letters
+                if (mUsername.getText().length() < 6) {
+                    mUsername.setError("Username must contain at least 6 letters");
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define `"Username must contain at least 6 letters"` on string resource

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: remove this line

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                if (mPassword.getText().length() < 6) {
+                    mPassword.setError("Password must contain at least 6 letters");
+                } else if (mPassword.getText().length() >= 6) {
+
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: remove this line

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                    if (!validatePassword(mPassword.getText().toString().trim())) {
+                        mPassword.setError("Password must contain at least one character and one number");
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define `"Password must contain at least one character and one number"` on string resource

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: remove this line

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+                if (!validateEmail(mEmail.getText().toString().trim())) {
+                    mEmail.setError("Invalid Email");
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define `"Invalid Email"` on string resource

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+        });
+
+        mRadioButton2.setOnClickListener(view -> {
+            mRadioChecked = true;
+            validation();
+        });
+    }
+
+    //validation() method checks all the necessary conditions in order for Button Apply to be enabled
+    private void validation() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: change method name more clearly

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+        if (mUsername.getText().length() >= 6
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: `6` should define as a constant

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+                && mPassword.getText().length() >= 6
+                && validatePassword(mPassword.getText().toString().trim())
+                && validateEmail(mEmail.getText().toString().trim())
+                && mRadioChecked == true) {
+
+            mApply.setEnabled(true);
+            mApply.setOnClickListener(view ->
+                    Toast.makeText(SignUpActivity.this, "Sign Up Successfully", Toast.LENGTH_LONG).show());
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define `"Sign Up Successfully"` on string resource

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+        } else {
+            mApply.setEnabled(false);
+        }
+    }
+
+    //This method checks if Password contains at least one character and one number
+    private boolean validatePassword(String password) {
+        String mPasswordPattern = "(?=.*?[a-z])(?=.*?[0-9]).{6,}";
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define as a global variable

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+        Pattern mPattern = Pattern.compile(mPasswordPattern);
+        Matcher mMatcher = mPattern.matcher(password);
+        return mMatcher.matches();
+    }
+
+    //This method checks if Email has the correct e-mail address format
+    private boolean validateEmail(String email) {
+        String mEmailPattern = "^[_A-Za-z0-9-]+(\\.[_A-Za-z0-9-]+)*@[A-Za-z0-9]+(\\.[A-Za-z0-9]+)*(\\.[A-Za-z]{2,})$";
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: define as a global variable

File: app/src/at-baole/res/layout/activity_sign_up.xml [hunk 15]
```diff
@@ -0,0 +1,136 @@
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+
```
Comment (at-vinhhuynh) [inline app/src/at-baole/res/layout/activity_sign_up.xml]: remove this line

File: app/src/main/res/layout/activity_main.xml [hunk 16]
```diff
@@ -1,8 +1,212 @@
...
+            <ImageView
+                android:layout_marginTop="10dp"
+                android:layout_weight="1"
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:src="@drawable/ic_dot"/>
+
+        </LinearLayout>
+
+        <LinearLayout
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: remove this layout

File: app/src/main/res/layout/activity_main.xml [hunk 17]
```diff
@@ -1,8 +1,212 @@
...
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="40dp"
+        android:layout_marginTop="150dp">
+
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="40dp"
+            android:text="Beatrice Maxwell"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: define `Beatrice Maxwell` on string resource

File: app/src/main/res/layout/activity_main.xml [hunk 18]
```diff
@@ -1,8 +1,212 @@
...
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="Slovakia"
+            android:fontFamily="casual"
+            android:textSize="16sp"/>
+
+    </LinearLayout>
+
+    <LinearLayout
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: remove this layout

File: app/src/main/res/layout/activity_main.xml [hunk 19]
```diff
@@ -1,8 +1,212 @@
...
+
+        <ImageView
+            android:layout_width="match_parent"
+            android:layout_height="5dp"
+            android:src="@drawable/ic_rectangle"/>
+
+    </RelativeLayout>
+
+    <GridView
+        android:id="@+id/gridView"
```
Comment (at-vinhhuynh) [inline app/src/main/res/layout/activity_main.xml]: change id more clearly

File: app/src/at-baole/res/values/strings.xml [hunk 20]
```diff
@@ -1,3 +1,13 @@
 <resources>
-    <string name="app_name">Intern18Summer</string>
+    <string name="app_name">SignUpActivity</string>
+    <string name="male">male</string>
+    <string name="female">female</string>
+    <string name="emailSub">Email subscriptions</string>
+    <string name="allowEmail">Allow email from other</string>
+    <string name="signUpActivity">SignUpActivity</string>
+    <string name="usernameError">Username must contain at least 6 letters</string>
+    <string name="passwordError1">Password must contain at least 6 letters</string>
```
Comment (at-vinhhuynh) [inline app/src/at-baole/res/values/strings.xml]: change resource name more clearly

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+        mRbMale.setOnClickListener(view -> {
+            isValidInformation();
+        });
+
+        mRbFemale.setOnClickListener(view -> {
+            isValidInformation();
+        });
+    }
+
+    public void checkUsernameLength() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: private

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+        if (mEdtUsername.getText().length() < MIN_LENGTH) {
+            mEdtUsername.setError(getString(R.string.username6Letters));
+            mEdtUsername.requestFocus();
+        } else {
+            mEdtUsername.setError(null);
+        }
+    }
+
+    public void checkPasswordLength() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: private

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+        if (mEdtPassword.getText().length() < MIN_LENGTH) {
+            mEdtPassword.setError(getString(R.string.password6Letters));
+        }
+    }
+
+    public void checkPasswordFormat() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: private

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+            if (!validatePassword(mEdtPassword.getText().toString().trim())) {
+                mEdtPassword.setError(getString(R.string.password1char1num));
+                mEdtPassword.requestFocus();
+            } else {
+                mEdtPassword.setError(null);
+            }
+        }
+    }
+
+    public void checkEmailFormat() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: private

File: app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+        if (!validateEmail(mEdtEmail.getText().toString().trim())) {
+            mEdtEmail.setError(getString(R.string.invalidEmail));
+            mEdtEmail.requestFocus();
+        } else {
+            mEdtEmail.setError(null);
+        }
+    }
+
+    public void onClickApplyButton() {
```
Comment (at-vinhhuynh) [inline app/src/at-baole/java/asiantech/internship/summer/SignUpActivity.java]: private

**Pull request merged**